### PR TITLE
Implement date.add

### PR DESCRIPTION
**Pull request by antepusic**
### Description

This PR introduces the `date.add` query function. Compatible with [`apoc.date.add`](https://neo4j.com/labs/apoc/4.3/overview/apoc.date/apoc.date.add/), it lets users do arithmetic with values that have user-specified temporal units.

### Pull request type

- [x] Algorithm/Module

### Related issues

#404 https://github.com/memgraph/memgraph/issues/1421

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Tests provided (unit / e2e)
- [ ] Code documentation
- [ ] README short description

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    > We introduced the `date.add` query function for more convenience in handling temporal values.
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: python/mage/date/unit_conversion.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    elif unit in Units.HOUR:
+        return datetime.timedelta(hours=time)
+    elif unit in Units.DAY:
+        return datetime.timedelta(days=time)
+    else:
+        raise TypeError(f"The unit {unit} is not correct.")
+
+
+def to_int(duration: datetime.timedelta, unit: str) -> int:
+    print(duration)
```
Comment (imilinovic) [inline python/mage/date/unit_conversion.py:22]: I think this is a debug line

File: python/date.py [hunk 2]
```diff
@@ -3,6 +3,14 @@
...
 
 from mage.date.constants import Conversion, Epoch
+from mage.date.unit_conversion import to_int, to_timedelta
+
+
+MILLISECOND = {"ms", "milli", "millis", "milliseconds"}
+SECOND = {"s", "second", "seconds"}
+MINUTE = {"m", "minute", "minutes"}
+HOUR = {"h", "hour", "hours"}
+DAY = {"d", "day", "days"}
```
Comment (imilinovic) [inline python/date.py:13]: This is unused, otherwise can be imported from mage.date.constants if needed

**Review by imilinovic — CHANGES_REQUESTED** [review]
small changes

File: python/mage/date/unit_conversion.py (hunk 1 above)
Comment (antepusic) [reply python/mage/date/unit_conversion.py]: Removed

File: python/date.py (hunk 2 above)
Comment (antepusic) [reply python/date.py]: This was a duplicate, thanks for pointing it out!

**Review by imilinovic — APPROVED** [review]
Looks good!
Just one more very minor thing, newline missing at `e2e/date_test/test_add_incorrect_second_unit/test.yml`

**Comment by antepusic** [comment]
@kgolubic Added a documentation PR, see the original message or the notification above this message.

**Pull request merged**